lib/src/rules/type_init_formals.dart

```diff
@@ -47,29 +48,46 @@ class TypeInitFormals extends LintRule {
   @override
   void registerNodeProcessors(
       NodeLintRegistry registry, LinterContext context) {
-    var visitor = _Visitor(this);
+    var visitor = _Visitor(this, context);
     registry.addFieldFormalParameter(this, visitor);
+    registry.addSuperFormalParameter(this, visitor);
   }
 }
 
 class _Visitor extends SimpleAstVisitor<void> {
+  final LinterContext context;
   final LintRule rule;
 
-  _Visitor(this.rule);
+  _Visitor(this.rule, this.context);
 
   @override
   void visitFieldFormalParameter(FieldFormalParameter node) {
     var nodeType = node.type;
-    if (nodeType != null) {
-      var cls = node.thisOrAncestorOfType<ClassDeclaration>()?.declaredElement;
-      if (cls != null) {
-        var field = cls.getField(node.identifier.name);
-        // If no such field exists, the code is invalid; do not report lint.
-        if (field != null) {
-          if (nodeType.type == field.type) {
-            rule.reportLint(nodeType);
-          }
-        }
+    if (nodeType == null) return;
+    var cls = node.thisOrAncestorOfType<ClassDeclaration>()?.declaredElement;
+    if (cls == null) return;
+
+    var field = cls.getField(node.identifier.name);
+    // If no such field exists, the code is invalid; do not report lint.
+    if (field != null) {
+      if (nodeType.type == field.type) {
+        rule.reportLint(nodeType);
+      }
+    }
+  }
+
+  @override
+  void visitSuperFormalParameter(SuperFormalParameter node) {
+    var nodeType = node.type;
+    if (nodeType == null) return;
+    var cls = node.thisOrAncestorOfType<ClassDeclaration>()?.declaredElement;
+    if (cls == null) return;
+
+    var inheritedElement = context.inheritanceManager
+        .getInherited2(cls, Name(null, node.identifier.name));
```

Comment:
For super-formal parameters we don't care about field, the super-constructor can do whatever it want with these parameters. But we inherit the type from the parameter of the super-constructor, see `SuperFormalParameterElement.superConstructorParameter`

Reply:
Ah.  That makes sense.  Updated.  PTAL. Thanks!